OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -139,45 +139,38 @@ static void BindHotkeyPref(HotkeyDefinition hd, HotkeyManager manager, Widget te
 			if (manager.GetFirstDuplicate(hd.Name, manager[hd.Name].GetValue(), hd) != null)
 				remapButton.GetColor = () => ChromeMetrics.Get<Color>("HotkeyColorInvalid");
 
-			remapButton.OnClick = () =>
+			var widgetArgs = new WidgetArgs
 			{
-				remapDialogRoot.RemoveChildren();
-
-				if (remapButton.IsHighlighted())
 				{
-					remapButton.IsHighlighted = () => false;
-
-					if (remapDialogPlaceholder != null)
-						remapDialogPlaceholder.Visible = true;
-
-					return;
-				}
+					"onSave", () =>
+					{
+						WidgetUtils.TruncateButtonToTooltip(remapButton, manager[hd.Name].GetValue().DisplayString());
+						remapButton.GetColor = () => ChromeMetrics.Get<Color>("ButtonTextColor");
+					}
+				},
+				{ "hotkeyDefinition", hd },
+				{ "hotkeyManager", manager },
+			};
 
-				if (remapDialogPlaceholder != null)
-					remapDialogPlaceholder.Visible = false;
+			if (isFirst)
+			{
+				remapDialogRoot.RemoveChildren();
+				remapButton.IsHighlighted = () => true;
+				Ui.LoadWidget("HOTKEY_DIALOG", remapDialogRoot, widgetArgs);
+			}
 
-				var siblings = parent.Children;
-				foreach (var sibling in siblings)
+			remapButton.OnClick = () =>
+			{
+				foreach (var sibling in parent.Children)
```

Comment:
Rather than manually walking all the siblings to manually toggle their state, define a `HotkeyDefinition selected` in the class, bind `button.IsHighlighted = () => selected == hd;` once during widget creation, then just update `selected` on select.

Reply:
This leads to CS0120 because `BindHotkeyPref` is a static method. Should I make the property or the method static to get around this?

Reply:
Make `BindHotkeyPref` not static instead!

Reply:
Done.